pkg/scheduler/clustereligibilitychecker/checker.go

```diff
@@ -0,0 +1,150 @@
+	clusterHealthCheckTimeout: defaultClusterHealthCheckTimeout,
+}
+
+// New returns a new cluster eligibility checker.
+func New(opts ...Option) *ClusterEligibilityChecker {
+	options := defaultCheckerOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+
+	return &ClusterEligibilityChecker{
+		clusterHeartbeatTimeout:   options.clusterHeartbeatTimeout,
+		clusterHealthCheckTimeout: options.clusterHealthCheckTimeout,
+	}
+}
+
+// IsEligible returns if a cluster is eligible for resource placement; if not, it will
+// also return the reason.
+func (checker *ClusterEligibilityChecker) IsEligible(cluster *fleetv1beta1.MemberCluster) (eligible bool, reason string) {
+	// Filter out clusters that have left the fleet.
+	if cluster.Spec.State == fleetv1beta1.ClusterStateLeave {
+		return false, "cluster has left the fleet"
+	}
+
+	// Note that the following checks are performed against one specific agent, i.e., the member
+	// agent, which is critical for the work orchestration related tasks in the fleet; non-related
+	// agents (e.g., networking) are not accounted for in this plugin.
+
+	// Filter out clusters that are no longer connected to the fleet, i.e., its heartbeat signals
+	// have stopped for a prolonged period of time.
+	memberAgentStatus := cluster.GetAgentStatus(fleetv1beta1.MemberAgent)
+	if memberAgentStatus == nil {
+		// The member agent has not updated its status with the hub cluster yet.
+		return false, "cluster is not connected to the fleet: member agent not online yet"
+	}
+
+	sinceLastHeartbeat := time.Since(memberAgentStatus.LastReceivedHeartbeat.Time)
+	if sinceLastHeartbeat > checker.clusterHeartbeatTimeout {
+		// The member agent has not sent heartbeat signals for a prolonged period of time.
+		//
+		// Note that this plugin assumes minimum clock drifts between clusters in the fleet.
+		return false, fmt.Sprintf("cluster is not connected to the fleet: no recent heartbeat signals (last received %.2f minutes ago)", sinceLastHeartbeat.Minutes())
+	}
+
+	memberAgentJoinedCond := cluster.GetAgentCondition(fleetv1beta1.MemberAgent, fleetv1beta1.AgentJoined)
+	if memberAgentJoinedCond == nil || memberAgentJoinedCond.Status != metav1.ConditionTrue {
+		// The member agent has not joined yet; i.e., some of the controllers have not been
+		// spun up.
+		//
+		// Note that here no generation check is performed, as
+		// a) the member cluster object spec is most of the time not touched after creation; and
+		// b) as long as the heartbeat signal does not timeout, a little drift in genrations
+		//    should not exclude a cluster from resource scheduling.
+		return false, "cluster is not connected to the fleet: member agent not joined yet"
+	}
+
+	memberAgentHealthyCond := cluster.GetAgentCondition(fleetv1beta1.MemberAgent, fleetv1beta1.AgentHealthy)
+	if memberAgentHealthyCond == nil {
+		// The health condition is absent.
+		return false, "cluster is not connected to the fleet: health condition not available"
```

Comment:
```suggestion
		return false, "cluster is not connected to the fleet: member agent health status is in the unknown"
```

to indicate it's member agent health status.